### PR TITLE
Validation tests for dotnet

### PR DESCRIPTION
**Pull request by rossgrambo**
Adds simple tests for dotnet.

Makes a small adjustment in python tests. Taking the stance that if we place `"Result": null`- then we should expect the result to be null.

### CODE REVIEW

File: libraryValidations/Python/test_json_validations.py [hunk 1]
```diff
@@ -119,7 +121,5 @@ def run_tests(self, test_key):
                 user = feature_flag_test[INPUTS_KEY].get(USER_KEY, None)
                 groups = feature_flag_test[INPUTS_KEY].get(GROUPS_KEY, [])
                 variant = feature_manager.get_variant(feature_flag_test[FEATURE_FLAG_NAME_KEY], TargetingContext(user_id=user, groups=groups))
-                if not variant:
-                    logger.error(f"Variant is None for {feature_flag_id}")
-                    assert False, failed_description
+
```
Comment (mrm9084) [inline libraryValidations/Python/test_json_validations.py:124]: Is there a reason you removed this assert? If you unindent it and replace `False` with `variant` it should work, see provider version.

File: Samples/NoFilters.tests.json [hunk 2]
```diff
@@ -28,7 +28,7 @@
             "Exception": "Invalid setting 'enabled' with value 'invalid' for feature 'InvalidEnabled'."
         },
         "Variant": {
-            "Result": null
+            "Exception": "Invalid setting 'enabled' with value 'invalid' for feature 'InvalidEnabled'."
```
Comment (mrm9084) [inline Samples/NoFilters.tests.json:31]: Did you run this test with python? I'm unsure it would produce this exact message.

**Review by mrm9084 — CHANGES_REQUESTED** [review]

File: libraryValidations/Python/test_json_validations.py (hunk 1 above)
Comment (rossgrambo) [reply libraryValidations/Python/test_json_validations.py:125]: My goal was to separate null variant from a variant whos configuration value was null. But I just updated this in a different way with my latest commit.

File: Samples/NoFilters.tests.json (hunk 2 above)
Comment (rossgrambo) [reply Samples/NoFilters.tests.json:31]: Turns out I was running the tests wrong. Latest commit has them working in .net and python

File: Samples/BasicVariant.tests.json [hunk 3]
```diff
@@ -6,7 +6,9 @@
             "Result": "false"
         },
         "Variant": {
-            "Result": "default"
+            "Result": {
+                "ConfigurationValue": "default"
```
Comment (zhiyuanliang-ms) [inline Samples/BasicVariant.tests.json:10]: @mrm9084 We should verify both variant name and configuration value.

Comment (mrm9084) [reply Samples/BasicVariant.tests.json:10]: @rossgrambo, can you make this change, so we don't have a huge merge conflict with each other.

File: Samples/ProviderTelemetryComplete.tests.json [hunk 4]
```diff
@@ -88,7 +88,7 @@
     {
         "FeatureFlagName": "Complete",
         "Inputs": {
-            "user": "Rachel",
+            "Users": "Rachel",
             "group": "beta"
```
Comment (mrm9084) [inline Samples/ProviderTelemetryComplete.tests.json:92]:
```suggestion
            "Groups": ["beta"]
```
This seems to be a bug from my PR.

**Review by mrm9084 — CHANGES_REQUESTED** [review]

File: Samples/ProviderTelemetryComplete.tests.json (hunk 4 above)
Comment (rossgrambo) [reply Samples/ProviderTelemetryComplete.tests.json:92]: Oh thanks- I'll resolve it

**Review by mrm9084 — APPROVED** [review]

**Pull request merged**